lib/service/service_test.go

```diff
@@ -1820,19 +1821,23 @@ func TestInitDatabaseService(t *testing.T) {
 			cfg.Databases.Enabled = test.enabled
 			cfg.Databases.Databases = test.databases
 
+			// This timeout should consider time to receive the event + shutdown
+			// time.
+			ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+			defer cancel()
+
+			var eg errgroup.Group
 			process, err := NewTeleport(cfg)
 			require.NoError(t, err)
-			t.Cleanup(func() {
-				require.NoError(t, process.Close())
-			})
 			require.NoError(t, process.Start())
-
-			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-			defer cancel()
+			eg.Go(func() error { return process.WaitForSignals(ctx, nil) })
 
 			if !test.expectErr {
 				_, err := process.WaitForEvent(ctx, TeleportReadyEvent)
 				require.NoError(t, err)
+				require.NoError(t, process.Close())
+				// Expect Teleport to shutdown without reporting any issue.
+				require.NoError(t, eg.Wait())
```

Comment:
Should this be a t.Cleanup so it runs even if the assertions before it fail?

(Same for the Wait in the "error" code branch.)

Ie:

```go
			t.Cleanup(func() {
				cancel()
				assert.NoError(t, eg.Wait(), "eg.Wait errored")
			})
```

Reply:
I agree that we should always call the `eg.Wait()` to ensure the Teleport state regardless of the assertions' results.

Just a brief context on this change: The main reason for moving this away from `t.Cleanup` was that the assertion on the `WaitForSignals` (which is on the errgroup) differs depending on the value on `test.expectErr`). But I guess having two `t.Cleanup` definitions per "test branch" is okay.

Reply:
The Cleanup doesn't need to care about the WaitForSignals, right? Just the cancel/Wait?

Reply:
Yes, I mean that the `WaitForSignals` is done on the errgroup goroutine so that it will be the error reported by `eg.Wait`